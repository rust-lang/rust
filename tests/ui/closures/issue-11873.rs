fn main() {
    let mut v = vec![1];
    let mut f = || v.push(2);
    let _w = v; //~ ERROR: cannot move out of `v`

    f();
}
