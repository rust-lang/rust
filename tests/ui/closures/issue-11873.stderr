error[E0505]: cannot move out of `v` because it is borrowed
  --> $DIR/issue-11873.rs:4:14
   |
LL |     let mut f = || v.push(2);
   |                 -- - borrow occurs due to use in closure
   |                 |
   |                 borrow of `v` occurs here
LL |     let _w = v;
   |              ^ move out of `v` occurs here
LL |
LL |     f();
   |     - borrow later used here

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0505`.
